app/ide-desktop/lib/dashboard/src/utilities/safeJsonParse.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * @file
+ *
+ * A utility function to safely parse a JSON string.
+ * returns the default value if the JSON string is invalid.
+ * Also provides a type for the parsed JSON.
+ */
+
+/**
+ * Safely parse a JSON string.
+ * Parse the JSON string and return the default value if the JSON string is invalid.
+ * Or if the parsed JSON does not match the type assertion.
+ */
+export function safeJsonParse<T = unknown>(
+  value: string,
+  defaultValue: T,
+  predicate: (parsed: unknown) => parsed is T
+): T {
+  try {
+    const parsed: unknown = JSON.parse(value)
+    return predicate(parsed) ? parsed : defaultValue
+  } catch {
+    return defaultValue
```

Comment:
Maybe we want to log if we're unable to parse the JSON for some reason? Just for debugging purposes, if we start to get `defaultValue` when we don't expect to.